src/providers/pkcs11_provider/mod.rs

```diff
@@ -113,7 +113,11 @@ impl Pkcs11Provider {
                         ) {
                             Ok(key_id) => key_id,
                             Err(response_status) => {
-                                error!("Error getting the KeyInfo for triple:\n{}\n(error: {}), continuing...", key_triple, response_status);
+                                if crate::utils::GlobalConfig::log_error_details() {
+                                    error!("Error getting the KeyInfo for triple:\n{}\n(error: {}), continuing...", key_triple, response_status);
+                                } else {
+                                    error!("Error getting the KeyInfo, continuing...");
+                                }
```

Comment:
I am thinking, to cater for this formatting, could we modify the `format_error` macro to be the following:
```rust
format_error!("Error finding key objects ({})", e);
```
and depending if permitted, the `()` will be expended with the error or left empty? That way we can place the error wherever we want in the format string?

Reply:
Unfortunately the first argument into `error!` (and all other macros) has to be a string literal. And also, it feels a bit weird to do string processing just to be able to use that macro everywhere